src/brain/operating_unit_util.cpp

```diff
@@ -0,0 +1,36 @@
+#include <cstdint>
+
+#include "brain/brain_defs.h"
+#include "brain/operating_unit_util.h"
+
+namespace terrier::brain {
+
+bool OperatingUnitUtil::IsOperatingUnitTypeBlocking(ExecutionOperatingUnitType f) {
+  switch (f) {
+    case ExecutionOperatingUnitType::HASHJOIN_BUILD:
+    case ExecutionOperatingUnitType::SORT_BUILD:
+    case ExecutionOperatingUnitType::AGGREGATE_BUILD:
+    case ExecutionOperatingUnitType::CREATE_INDEX:
+      return true;
+    default:
+      return false;
+  }
+}
+
+ExecutionOperatingUnitType OperatingUnitUtil::GetNonParallelType(ExecutionOperatingUnitType f) {
+  switch (f) {
+    case ExecutionOperatingUnitType::PARALLEL_MERGE_HASHJOIN:
+      return ExecutionOperatingUnitType::HASHJOIN_BUILD;
+    case ExecutionOperatingUnitType::PARALLEL_MERGE_AGGBUILD:
+      return ExecutionOperatingUnitType::AGGREGATE_BUILD;
+    case ExecutionOperatingUnitType::PARALLEL_SORT_STEP:
+    case ExecutionOperatingUnitType::PARALLEL_SORT_MERGE_STEP:
+      return ExecutionOperatingUnitType::SORT_BUILD;
+    case ExecutionOperatingUnitType::CREATE_INDEX_MAIN:
+      return ExecutionOperatingUnitType::CREATE_INDEX;
+    default:
+      return ExecutionOperatingUnitType::INVALID;
```

Comment:
This function sort of serves a dual purpose: (1) checking if the ExecutionOperatingUnitType represents a parallel OU and (2) returning the non-parallel type if it is a parallel OU. So with the current way the function is written, an assert can't be inserted since the function is supposed to test.